rasa/nlu/server.py

```diff
@@ -31,324 +52,276 @@ def create_argument_parser():
+        return response.json(
             {
                 "version": rasa.__version__,
                 "minimum_compatible_version": MINIMUM_COMPATIBLE_VERSION,
             }
         )
 
-    @app.route("/status", methods=["GET", "OPTIONS"])
-    @requires_auth
-    @check_cors
-    def status(self, request):
-        request.setHeader("Content-Type", "application/json")
-        return json_to_string(self.data_router.get_status())
+    @app.get("/status")
+    @requires_auth(app, token)
+    async def status(request):
+        return response.json(data_router.get_status())
 
-    def extract_json(self, content):
+    def extract_json(content):
         # test if json has config structure
         json_config = simplejson.loads(content).get("data")
 
         # if it does then this results in correct format.
         if json_config:
-
-            model_config = simplejson.loads(content)
-            data = json_config
+            return simplejson.loads(content), json_config
 
         # otherwise use defaults.
         else:
+            return default_model_config, content
 
-            model_config = self.default_model_config
-            data = content
+    def extract_data_and_config(request):
 
-        return model_config, data
-
-    def extract_data_and_config(self, request):
+        request_content = request.body.decode("utf-8", "strict")
 
-        request_content = request.content.read().decode("utf-8", "strict")
-        content_type = self.get_request_content_type(request)
-
-        if "yml" in content_type:
+        if "yml" in request.content_type:
             # assumes the user submitted a model configuration with a data
             # parameter attached to it
 
             model_config = rasa.utils.io.read_yaml(request_content)
             data = model_config.get("data")
 
-        elif "json" in content_type:
-
-            model_config, data = self.extract_json(request_content)
+        elif "json" in request.content_type:
+            model_config, data = extract_json(request_content)
 
         else:
-
             raise Exception(
                 "Content-Type must be 'application/x-yml' or 'application/json'"
             )
 
         return model_config, data
 
-    def get_request_content_type(self, request):
-        content_type = request.requestHeaders.getRawHeaders("Content-Type", [])
-
-        if len(content_type) is not 1:
-            raise Exception("The request must have exactly one content type")
-        else:
-            return content_type[0]
-
-    @app.route("/train", methods=["POST", "OPTIONS"], branch=True)
-    @requires_auth
-    @check_cors
-    @inlineCallbacks
-    def train(self, request):
+    @app.post("/train")
+    @requires_auth(app, token)
+    async def train(request):
         # if not set will use the default project name, e.g. "default"
-        project = parameter_or_default(request, "project", default=None)
+        project = request.raw_args.get("project", None)
```

Comment:
apparently `raw_args` gets deprecated in future version. Maybe use `args` ? https://sanic.readthedocs.io/en/latest/sanic/request_data.html